### ISSUE TITLE
Compile error on osx

### ISSUE THREAD

**Issue by agoering** [gevero/py_gmm#14]
Hi Giovanni, I am having trouble installing py_gmm on macOS 10.13. 

Error message: https://gist.github.com/agoering/78a5d7592206eb7bb21f6d2925f9d041

At first it seemed like it wasn't finding gfortran, so I changed the location that f2py.sh looks for it from /usr/bin/gfortran to /usr/local/bin/gfortran. When that failed I tried figuring out the error myself; it seemed like openblas was missing so I took these steps:

brew install openblas
export LDFLAGS="-L/usr/local/opt/openblas/lib" && export CPPflags="-I/usr/local/opt/openblas/include" && export CPPflags="-I/usr/local/opt/openblas/include"

This didn't fix the issue with this package, but it interestingly did fix issues I had with installing pytmatrix. The py_gmm error was the same. I then found the closed issue that mentioned a bug in anaconda which fixed it for a linux user, and so I followed the stackexchange instructions to update conda and anaconda via:

conda update conda && conda update python
conda update anaconda

However, it is still throwing the same error.

I would appreciate any insight you might have!

**Comment by gevero** [comment]
Hi Andrea

Please revert all  the changes you made (brew install openblas, export...) and do 'conda install openblas' and see if it works.

best

Giovanni

**Comment by agoering** [comment]
Hi Giovanni, thanks for the quick reply. I did the following:

brew uninstall openblas
conda install openblas
unset $CPPFLAGS
unset $LDFLAGS
unset $PKG_CONFIG_PATH

Still no dice. The error message has a small difference, I added that to the [gist](https://gist.github.com/agoering/78a5d7592206eb7bb21f6d2925f9d041).

I have to admit I am kind of groping in the dark with respect to the environment variables... I only know enough about bash to be a danger to myself and my computer.

**Comment by gevero** [comment]
Hi Andrea

if you look at your first gist at [line 356](https://gist.github.com/agoering/78a5d7592206eb7bb21f6d2925f9d041#file-py_gmm-osx-compile-error-L356), you'll see that  during the compilation you are not invoking anaconda3, but some kind of local python installation in `/usr/local/lib`. I don't know why is that, nevertheless I would clean up your system, and install anaconda in python 3 version. In alternative, maybe that you did not install `numpy` from anaconda, and thus you are calling the system version of `f2py`. Once you fix these issues the installation should work.

**Comment by gevero** [comment]
Hi Andrea

If you have no further questions or observations, I am going to close the issue.

Best 

Giovanni

**Comment by agoering** [comment]
Hi Giovanni, I haven't had time to work on the issue for a few days now.
I'm in the process of cleaning up my machine as you suggested. If it fails
completely I'll try from on a different Mac and if I continue to have
issues I will let you know. Thanks for your help!

**Comment by agoering** [comment]
Hi Giovanni, I finally got everything to work. I had some challenges with the notebook importing py_gmm and thought I would leave some feedback in case others have the same problem. Thanks for all your help and I can't wait to get going with some simulations!

**Problems and solutions:**
1. Numpy warning (**unresolved but inconsequential**)
I made a new python3 environment and installed openblas with conda and the program compiled (with the same deprecated numpy error). I tried to fix the numpy warning by installing the channel using `conda install -c anaconda numpy` and this did not change the numpy warning but the program compiles.

2. Import error for libgfortran and libopenblas (**RESOLVED**)
I launched the Au_dimer_scattering example via `jupyter notebook --NotebookApp.iopub_data_rate_limi1.0e10` and ran the initialization cell which didn't import py_gmm and resulted in the following import error:

> ImportError: dlopen(../py_gmm/gmm_py.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libgfortran.2.dylib
  Referenced from: /Users/ayocom/anaconda/envs/python3/lib/libopenblas_nehalemp-r0.2.19.dylib
  Reason: image not found

I tried `conda install -c conda-forge openblas` based on info about [openblas and libgfortran from anaconda not working](https://github.com/conda-forge/staged-recipes/issues/3248#issuecomment-315505112).

After recompiling, the notebook worked!

**Versions of relevant packages in the working conda environment**

jupyter                   1.0.0            py36h598a6cc_0  
libgfortran               3.0.1                h93005f0_2
numpy                     1.13.3           py36h2cdce51_0    anaconda
openblas                  0.2.20                        2    conda-forge
python                    3.6.3                h6804ab2_0
